shell/app/modules/project/pages/homepage/readme-markdown.tsx

```diff
@@ -0,0 +1,96 @@
+import React from 'react';
+import i18n from 'core/i18n';
+import { Tooltip } from 'antd';
+import { MarkdownEditor } from 'common';
+import remarkGfm from 'remark-gfm';
+import { useUpdate } from 'common/use-hooks';
+import ReactMarkdown from 'react-markdown';
+import { ScalableImage } from 'common/components/edit-field';
+import './index.scss';
+
+interface IMdProps {
+  value?: string;
+  originalValue?: string;
+  disabled?: boolean;
+  hasEdited?: boolean;
+  maxHeight: number;
+  onChange: (v: string) => void;
+  onSave: (v?: string) => void;
+}
+export const ReadMeMarkdown = ({ value, onChange, onSave, disabled, originalValue, maxHeight, ...rest }: IMdProps) => {
```

Comment:
1. seems `rest` has just one prop `hasEdited `, no need deconstruct it
2. is `hasEdited` a `MarkdownEditor` prop?